pennylane/tape/tapes/qubit_param_shift.py

```diff
@@ -169,7 +169,13 @@ def processing_fn(results):
                 array[float]: 1-dimensional array of length determined by the tape output
                 measurement statistics
             """
-            return np.dot(coeffs, np.squeeze(results))
+            results = np.squeeze(results)
+
+            if results.dtype is not np.dtype("O"):
+                dot = lambda x: np.dot(coeffs, x)

```

Comment:
This would not work with non-ragged arrays?

Reply:
Nope, `np.dot` is doing some really weird broadcasting that I can not interpret. The returned array has the complete incorrect shape.